02-peer-protocol.md

```diff
@@ -787,9 +787,17 @@ is destined, is described in [BOLT #4](04-onion-routing.md).
 #### Requirements
 
 A sending node:
-  - MUST NOT offer `amount_msat` it cannot pay for in the
-remote commitment transaction at the current `feerate_per_kw` (see "Updating
-Fees") while maintaining its channel reserve.
+  - if it is _responsible_ for paying the Bitcoin fee:
+    - MUST NOT offer `amount_msat` it cannot pay for in the remote commitment
+    transaction at the current `feerate_per_kw` (see "Updating Fees") while
+    maintaining its channel reserve.
+    - MUST NOT offer `amount_msat` if the resulting balance doesn't allow it to
```

Comment:
is "resulting balance" here referring to balance after the HTLC is settled or while it is still on the commitment?

Reply:
This is where I struggled the most with the wording...

It's both in my opinion, but unless I'm missing something (correct me if I'm wrong there, this really needs some challenging) the more restrictive case is the one where the HTLC is still in the commitment tx (because the commit tx fee is higher in that case). If you're able to satisfy the condition while the HTLC is in the commitment, you're able to satisfy it once it's resolved.

If we agree that's what we expect, I can try to find a more accurate wording to reflect that. Maybe "MUST NOT offer `amount_msat` if, once that HTLC is added to the remote commitment transaction, the sender cannot pay the fee for a future additional HTLC at `2*feerate_per_kw` while maintaining its channel reserve."?

Reply:
Mhm, maybe we have to be this restrictive, otherwise the other node possibly cannot add more HTLCs while this HTLC is still pending. If we only required that the balance was high enough after the HTLC had been settled, then the channel would essentially be "locked" while this HTLC was still pending. But it would also make less of you balance available to send of course...

Reply:
Done, let me know if the current wording is better.